flow_prompt/ai_models/claude/claude_model.py

```diff
@@ -0,0 +1,180 @@
+    sonnet = "Claude 3 Sonnet"
+    opus = "Claude 3 Opus"
+
+DEFAULT_PRICING = {
+    "price_per_prompt_1k_tokens": Decimal(0.00025),
+    "price_per_sample_1k_tokens": Decimal(0.00125),
+}
+
+CLAUDE_AI_PRICING = {
+    FamilyModel.haiku.value: {
+        C_200K: {
+            "price_per_prompt_1k_tokens": Decimal(0.00025),
+            "price_per_sample_1k_tokens": Decimal(0.00125),
+        }
+    },
+    FamilyModel.sonnet.value: {
+        C_200K: {
+            "price_per_prompt_1k_tokens": Decimal(0.003),
+            "price_per_sample_1k_tokens": Decimal(0.015),
+        }
+    },
+    FamilyModel.opus.value: {
+        C_200K: {
+            "price_per_prompt_1k_tokens": Decimal(0.015),
+            "price_per_sample_1k_tokens": Decimal(0.075),
+        }
+    },
+}
+
+
+@dataclass(kw_only=True)
+class ClaudeAIModel(AIModel):
+    model: str
+    api_key: str = ""
+    provider: AI_MODELS_PROVIDER = AI_MODELS_PROVIDER.CLAUDE
+    family: str = None
+    
+    def __post_init__(self):
+        if 'haiku' in self.model:
+            self.family = FamilyModel.haiku.value
+        elif 'sonnet' in self.model:
+            self.family = FamilyModel.sonnet.value
+        elif 'opus' in self.model:
+            self.family = FamilyModel.opus.value
+        else:
+            logger.warning(
+                f"Unknown family for {self.model}. Please add it obviously. Setting as Claude 3 Haiku"
+            )
+            self.family = FamilyModel.haiku.value
+
+        logger.debug(f"Initialized ClaudeAIModel: {self}")
+        
+        secrets = Secrets()
+        self.api_key = secrets.CLAUDE_API_KEY
+        
+
+    def call(self, messages: t.List[dict], max_tokens: int, **kwargs) -> AIResponse:
+        
+        common_args = {
+            "top_p": 1,
+            "temperature": 0,
+            "max_tokens": max_tokens,
+            "stream": False,
+        }
+        kwargs = {
+            **{
+                "messages": messages,
+            },
+            **common_args,
+            **self.get_params(),
+            **kwargs,
+        }
+        
+        logger.debug(
+            f"Calling {messages} with max_tokens {max_tokens} and kwargs {kwargs}"
+        )
+        
+        stream_function = kwargs.get('stream_function')
+        check_connection = kwargs.get('check_connection')
+        stream_params = kwargs.get('stream_params')
+        
+        content = ""
+        
+        try:
+            if kwargs.get('stream'):
+                with anthropic.Anthropic(api_key=self.api_key).messages.stream(
+                    model=self.model,
+                    max_tokens=max_tokens,
+                    messages=messages
+                ) as stream:
+                    idx = 0 
+                    for text in stream.text_stream:
+                        if idx % 5 == 0:
+                            if not check_connection(**stream_params):
+                                raise ConnectionError  
```

Comment:
can you please raise always with message? raise Error('message')
never like raise Error